src/test/unit/ilp.test.js

```diff
@@ -68,3 +68,51 @@ describe('ILP', () => {
         expect(valid).toBeTruthy();
     });
 });
+
+describe('Ilp Packet Decoding and Validation', () => {
+    let ilp;
+    let ilpCombo;
+    const transferRequest = require('./data/transferRequest');
+
+    beforeEach(() => {
+        ilp = new Ilp({secret: 'test'});
+        ilpCombo = ilp.getQuoteResponseIlp(quoteRequest, partialResponse);
+        transferRequest.ilpPacket = ilpCombo.ilpPacket;
+        transferRequest.condition = ilpCombo.condition;
+    });
+
+    test('Should decode the IlpPacket', () => {
+        const decodedIlp = ilp.decodeIlpPacket(ilpCombo.ilpPacket);
+        
+        expect(decodedIlp).toBeTruthy();
+        expect(decodedIlp).toHaveProperty('amount');
+        expect(decodedIlp).toHaveProperty('account');
+        expect(decodedIlp).toHaveProperty('data');
+    });
+
+    test('Should generate transaction object from an Ilp packet', () => {
+        const transactionObject = ilp.getTransactionObject(ilpCombo.ilpPacket);
+        
+        expect(transactionObject).toBeTruthy();
+        expect(transactionObject).toHaveProperty('transactionId');
+        expect(transactionObject).toHaveProperty('quoteId');
+        expect(transactionObject).toHaveProperty('payee');
+        expect(transactionObject).toHaveProperty('payer');
+        expect(transactionObject).toHaveProperty('amount');
+        expect(transactionObject).toHaveProperty('transactionType');
```

Comment:
Probably better to just compare to the `transferRequest` object with
```
expect(transactionObject).toMatchObject(transferRequest);
```

Reply:
I think that's not possible to match this object to neither transferRequest nor quoteRequest.
The transactionObject is a constructed object based on quoterequest and quoteresponse.